pkg/util/errors/error.go

```diff
@@ -0,0 +1,40 @@
+package errors
+
+import (
+	"errors"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type GRPCError struct {
+	internalError error
+	code          codes.Code
+}
+
+func New(inner error, code ...codes.Code) *GRPCError {
+	return &GRPCError{
+		internalError: inner,
+		code: func() codes.Code {
+			if len(code) == 0 {
+				return codes.Unknown
+			}
+			return code[0]
+		}(),
+	}
+}
+
+func (e *GRPCError) Error() string {
+	return e.internalError.Error()
+}
+
+func (e *GRPCError) GRPCStatus() *status.Status {
+	return status.New(e.code, e.internalError.Error())
```

Comment:
You might also need to check if e.internalError implements `interface{ GRPCStatus() *Status }` (or maybe check if it's `*status.Error`, not sure off the top of my head). Either way, make sure you don't accidentally lose `Details` from the original message